File: ios-native/Sources/ALCore/Mascot/MascotIDs.swift
// Port of `src/mascot/ids.ts`.
//
// D8 — this lives in ALCore, not ALArt: the shop (ALUI), the persisted profile
// (Persistence) and the rigs (ALArt) all read these strings.
//
// spec/mascot.md §12 — EVERY STRING IN THIS FILE IS A PERSISTENCE CONTRACT.
// The slot names land in `MascotConfig.colors[slot]` / `.styles[slot]` and the
// accessory ids land in `MascotConfig.accessories`, both of which are written to
// disk and pushed across the sync wire. Renaming one "for Swift style" orphans
// every existing profile at migration time — exactly the rule D11 froze for
// `ExerciseId` raw values. The Swift *identifiers* below may be idiomatic; the
// *string values* may not change. `MascotIDStabilityTests` pins all 54 of them
// — 17 colour slots, 12 style slots, 25 accessory ids — against a hand-written
// oracle so a rename fails loudly.

/**
 * Single source of truth for the config keys + option ids the rig reads.
 * Both Mascot.tsx (reader) and catalog.ts (writer) import from here so the
 * slot / value / accessory-id strings can never drift apart.
 *
 * Owned by AGENT A.
 */
// NB: the TS is one nested `as const` object per kind, keyed by species. Swift
// gets one caseless enum per (kind, species) because the species do not share a
// slot set — the unicorn has a horn and a mane, the fox has a tail tip. A
// `[Species: [String: String]]` dictionary would compile but would turn
// `COLOR_SLOT.unicorn.body`, a typo the TS compiler catches, into a runtime nil.

/// Colour config slots per species — written to config.colors[slot].
public enum ColorSlot {
    public enum Unicorn {
        public static let body = "bodyColor"
        public static let horn = "hornColor"
        public static let mane = "maneColor"
        public static let tail = "tailColor"
    }
    public enum Cat {
        public static let body = "bodyColor"
        public static let belly = "bellyColor"
        public static let tail = "tailColor"
    }
    public enum Fox {
        public static let body = "bodyColor"
        public static let belly = "bellyColor"
        public static let tailTip = "tailTipColor"
    }
    public enum Rabbit {
        public static let body = "bodyColor"
        public static let belly = "bellyColor"
        public static let inner = "innerEarColor"
    }
    public enum Dragon {
        public static let body = "bodyColor"
        public static let belly = "bellyColor"
        public static let wing = "wingColor"
        public static let horn = "hornColor"
    }
}

/// Style config slots per species — written to config.styles[slot].
public enum StyleSlot {
    public enum Unicorn {
        public static let tail = "tailStyle"
        public static let horn = "hornStyle"
    }
    public enum Cat {
        public static let hair = "hair"
        public static let tail = "tailSize"
    }
    public enum Fox {
        public static let fur = "furPattern"
        public static let tail = "tailSize"
    }
    public enum Rabbit {
        public static let ear = "earStyle"
        public static let tail = "tailStyle"
        public static let fur = "furPattern"
    }
    public enum Dragon {
        public static let horn = "hornStyle"
        public static let crest = "crestStyle"
        public static let tail = "tailStyle"
    }
}

/// Accessory option ids — matched verbatim against config.accessories.
public enum Accessory {
    public enum Unicorn {
        public static let ribbon = "unicorn.accessory.ribbon"
        public static let flowerCrown = "unicorn.accessory.flower-crown"
        public static let starClip = "unicorn.accessory.star-clip"
        public static let swimsuit = "unicorn.accessory.swimsuit"
        public static let swimRing = "unicorn.accessory.swim-ring"
    }
    public enum Cat {
        public static let bow = "cat.accessory.bow"
        public static let bellCollar = "cat.accessory.bell-collar"
        public static let partyHat = "cat.accessory.party-hat"
        public static let swimsuit = "cat.accessory.swimsuit"
        public static let swimRing = "cat.accessory.swim-ring"
    }
    public enum Fox {
        public static let scarf = "fox.accessory.scarf"
        public static let beanie = "fox.accessory.beanie"
        public static let boots = "fox.accessory.boots"
        public static let swimsuit = "fox.accessory.swimsuit"
        public static let swimRing = "fox.accessory.swim-ring"
    }
    public enum Rabbit {
        public static let bow = "rabbit.accessory.bow"
        public static let nightcap = "rabbit.accessory.nightcap"
        public static let stardust = "rabbit.accessory.stardust"
        public static let swimsuit = "rabbit.accessory.swimsuit"
        public static let swimRing = "rabbit.accessory.swim-ring"
    }
    // No swim pair on the dragon — the cross-species tradition is deliberately
    // broken here (user decision, wardrobe v2): a fire dragon doesn't bathe.
    public enum Dragon {
        public static let cape = "dragon.accessory.cape"
        public static let goggles = "dragon.accessory.goggles"
        public static let fang = "dragon.accessory.fang-necklace"
        public static let treasure = "dragon.accessory.treasure"
        public static let blueFlame = "dragon.accessory.blue-flame"
    }
}

File: apps/game-android/core/src/main/kotlin/fr/dappit/attrapelettres/core/mascot/MascotIds.kt
package fr.dappit.attrapelettres.core.mascot

// Port of `src/mascot/ids.ts`.
//
// This lives in :core, not :art: the shop (:ui), the persisted profile
// (persistence/) and the rigs (:art) all read these strings.
//
// EVERY STRING IN THIS FILE IS A PERSISTENCE CONTRACT. The slot names land in
// `MascotConfig.colors[slot]` / `.styles[slot]` and the accessory ids land in
// `MascotConfig.accessories`, both of which are written to disk and pushed
// across the sync wire. Renaming one "for Kotlin style" orphans every existing
// profile at migration time — exactly the rule A2 froze for `ExerciseId` wire
// values. The Kotlin *identifiers* below may be idiomatic; the *string values*
// may not change. `MascotCatalogTest` pins all 54 of them — 17 colour slots,
// 12 style slots, 25 accessory ids — against a hand-written oracle so a rename
// fails loudly.

/**
 * Single source of truth for the config keys + option ids the rig reads.
 * Both Mascot.tsx (reader) and catalog.ts (writer) import from here so the
 * slot / value / accessory-id strings can never drift apart.
 *
 * Owned by AGENT A.
 */
// NB: the TS is one nested `as const` object per kind, keyed by species. Kotlin
// gets one nested object per (kind, species) because the species do not share a
// slot set — the unicorn has a horn and a mane, the fox has a tail tip. A
// `Map<Species, Map<String, String>>` would compile but would turn
// `COLOR_SLOT.unicorn.body`, a typo the TS compiler catches, into a runtime null.

/** Colour config slots per species — written to config.colors[slot]. */
object ColorSlot {
    object Unicorn {
        const val body = "bodyColor"
        const val horn = "hornColor"
        const val mane = "maneColor"
        const val tail = "tailColor"
    }

    object Cat {
        const val body = "bodyColor"
        const val belly = "bellyColor"
        const val tail = "tailColor"
    }

    object Fox {
        const val body = "bodyColor"
        const val belly = "bellyColor"
        const val tailTip = "tailTipColor"
    }

    object Rabbit {
        const val body = "bodyColor"
        const val belly = "bellyColor"
        const val inner = "innerEarColor"
    }

    object Dragon {
        const val body = "bodyColor"
        const val belly = "bellyColor"
        const val wing = "wingColor"
        const val horn = "hornColor"
    }
}

/** Style config slots per species — written to config.styles[slot]. */
object StyleSlot {
    object Unicorn {
        const val tail = "tailStyle"
        const val horn = "hornStyle"
    }

    object Cat {
        const val hair = "hair"
        const val tail = "tailSize"
    }

    object Fox {
        const val fur = "furPattern"
        const val tail = "tailSize"
    }

    object Rabbit {
        const val ear = "earStyle"
        const val tail = "tailStyle"
        const val fur = "furPattern"
    }

    object Dragon {
        const val horn = "hornStyle"
        const val crest = "crestStyle"
        const val tail = "tailStyle"
    }
}

/** Accessory option ids — matched verbatim against config.accessories. */
object Accessory {
    object Unicorn {
        const val ribbon = "unicorn.accessory.ribbon"
        const val flowerCrown = "unicorn.accessory.flower-crown"
        const val starClip = "unicorn.accessory.star-clip"
        const val swimsuit = "unicorn.accessory.swimsuit"
        const val swimRing = "unicorn.accessory.swim-ring"
    }

    object Cat {
        const val bow = "cat.accessory.bow"
        const val bellCollar = "cat.accessory.bell-collar"
        const val partyHat = "cat.accessory.party-hat"
        const val swimsuit = "cat.accessory.swimsuit"
        const val swimRing = "cat.accessory.swim-ring"
    }

    object Fox {
        const val scarf = "fox.accessory.scarf"
        const val beanie = "fox.accessory.beanie"
        const val boots = "fox.accessory.boots"
        const val swimsuit = "fox.accessory.swimsuit"
        const val swimRing = "fox.accessory.swim-ring"
    }

    object Rabbit {
        const val bow = "rabbit.accessory.bow"
        const val nightcap = "rabbit.accessory.nightcap"
        const val stardust = "rabbit.accessory.stardust"
        const val swimsuit = "rabbit.accessory.swimsuit"
        const val swimRing = "rabbit.accessory.swim-ring"
    }

    // No swim pair on the dragon — the cross-species tradition is deliberately
    // broken here (user decision, wardrobe v2): a fire dragon doesn't bathe.
    object Dragon {
        const val cape = "dragon.accessory.cape"
        const val goggles = "dragon.accessory.goggles"
        const val fang = "dragon.accessory.fang-necklace"
        const val treasure = "dragon.accessory.treasure"
        const val blueFlame = "dragon.accessory.blue-flame"
    }
}
